stages/org.osbuild.oscap-remediation

```diff
@@ -0,0 +1,107 @@
+Execute oscap remediation on first boot
+
+The openscap scanner package uses systemd's  `system-update.target` to run an openscap
+remediation at first boot[1]. The `openscap-remediate.service` expects a `/system-update`
+symlink that points to an openscap offline remediation config file. The requirement
+of the config file is to have the basename of `oscap-remediate-offline.conf.sh` and
+must be accessible at boot time.
+
+This stage generates the openscap offline remediation config file with the required
+configurations. A `/system-update` symlink is then created which points to the
+config file.
+
+Once executed, the symlink will be removed to avoid an invocation loop. The oscap
+offline remediation configuration will be left intact.
+
+[1] https://github.com/OpenSCAP/openscap/blob/maint-1.3/docs/manual/manual.adoc#15-scanning-and-remediating-the-system-at-boot-time
+"""
+
+
+import os
+import sys
+
+import osbuild.api
+
+
+SCHEMA = """
+"additionalProperties": false,
+"required": ["profile_id", "datastream"],
+"properties": {
+  "profile_id": {
+    "type": "string",
+    "description": "The scap profile id"
+  },
+  "datastream": {
+    "type": "string",
+    "description": "The path to the datastream file"
+  },
+  "datastream_id": {
+    "type": "string",
+    "description": "The datastream id"
+  },
+  "xccdf_id": {
+    "type": "string",
+    "description": "The xccdf id"
+  },
+  "benchmark_id": {
+    "type": "string",
+    "description": "The benchmark id"
+  },
+  "tailoring": {
+    "type": "string",
+    "description": "The path to the custom tailoring file"
+  },
+  "tailoring_id": {
+    "type": "string",
+    "description": "The tailoring id"
+  },
+  "arf_result": {
+    "type": "string",
+    "description": "Filename and path for saving the arf results"
+  },
+  "html_report": {
+    "type": "string",
+    "description": "Filename and path for saving the html report"
+  },
+  "verbose_log": {
+    "type": "string",
+    "description": "Filename and path for verbose error messages"
+  },
+  "verbose_level": {
+    "type": "string",
+    "enum": ["DEVEL", "INFO", "ERROR", "WARNING"],
+    "description": "The verbosity level for the log messages",
+    "default": "INFO"
+  }
+}
+"""
+
+def main(tree, options):
+    config = {
+      "OSCAP_REMEDIATE_DS": options.get('datastream'),
+      "OSCAP_REMEDIATE_PROFILE_ID": options.get('profile_id'),
+      "OSCAP_REMEDIATE_DATASTREAM_ID": options.get('datastream_id'),
+      "OSCAP_REMEDIATE_XCCDF_ID": options.get('xccdf_id'),
+      "OSCAP_REMEDIATE_BENCHMARK_ID": options.get('benchmark_id'),
+      "OSCAP_REMEDIATE_TAILORING": options.get('tailoring'),
+      "OSCAP_REMEDIATE_TAILORING_ID": options.get('tailoring_id'),
+      "OSCAP_REMEDIATE_ARF_RESULT": options.get('arf_result'),
+      "OSCAP_REMEDIATE_HTML_REPORT":  options.get('html_report'),
+      "OSCAP_REMEDIATE_VERBOSE_LOG":  options.get('verbose_log'),
+      "OSCAP_REMEDIATE_VERBOSE_LEVEL": options.get('verbose_level')
```

Comment:
This will end up being `OSCAP_REMEDIATE_...=None` for values that are not set. You could split the map out as:
```python3
OSCAP_OPTIONS = {
  "datastream": "OSCAP_REMEDIATE_DS",
  "profile_id": "OSCAP_REMEDIATE_PROFILE_ID",
  ...
}
```

And then walk over the `options` or if you wanna enforce a stable order of items over `OSCAP_OPTIONS` and only write the ones that set, like:
```
for ours, theirs in OSCAP_OPTIONS.items():
   value = options.get(ours)
   if value:
     f.write(f"{theirs}={value}\n")
```
NB: this will also get you a final newline which the other code did not. Maybe not a bad idea.

Reply:
I was filtering out the options that were set to none in the list comprehension. But not a very readable
solution at all. So I will refactor that